alternator test_list_streams_limit_parameter: Got unexpected ARN values by list streams paged responses

Seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/181/testReport/alternator_stream_tests/TestAlternatorStreams/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split005___test_list_streams_limit_parameter/
Scylla 0d64f327e1af
```
self = <alternator_stream_tests.TestAlternatorStreams object at 0x7fc78f183b10>

    def test_list_streams_limit_parameter(self):
        """
        Test the list_streams command limit parameter.
        See that when the response is large enough, it can be paged
        correctly according the 'limit' value.
        Test steps:
        1. create and enable streams for 20 tables.
        2. read variable size chunks of these tables streams in list_streams command via random node.
        3. where using and verifying different 'limit' values + its total output.
        """
        stream_specification = enums.StreamSpecification.KEYS_ONLY.value
        self.prepare_dynamodb_cluster(num_of_nodes=3)
        node1 = self.cluster.nodelist()[0]
        table_names = [TABLE_NAME+str(idx) for idx in range(20)]
        for table_name in table_names:
            self.create_table(node=node1, table_name=table_name, stream_specification=stream_specification)
        tables_arns = [self.wait_for_active_stream(node=node1, table_name=table_name)[0] for table_name in table_names]
        total_limited_stream_responses = []
        last_evaluated_stream_arn = None
        for limit in [5, 7, 3, 1, 4]:  # slice the created 20 tables to various size chunks
            params = {'Limit': limit}
            if last_evaluated_stream_arn:
                params['ExclusiveStartStreamArn'] = last_evaluated_stream_arn
            dynamodb_api = self.get_dynamodb_api(node=random.choice(self.cluster.nodelist()))
            result = dynamodb_api.stream.list_streams(**params)
            assert len(result['Streams']) == limit, \
                f"Got unexpected number of streams [{len(result['Streams'])}] for [{limit}] requested!"
            total_limited_stream_responses += result['Streams']
            last_evaluated_stream_arn = result['LastEvaluatedStreamArn']
>       assert sorted(tables_arns) == sorted([stream['StreamArn'] for stream in total_limited_stream_responses]), \
            f"Got unexpected ARN values by list streams paged responses: {total_limited_stream_responses}"
E       AssertionError: Got unexpected ARN values by list streams paged responses: [{'StreamArn': 'Sea454df3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table19', 'StreamLabel': '"2023-01-21T05:53:23Z"'}, {'StreamArn': 'Se8c14603-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table17', 'StreamLabel': '"2023-01-21T05:53:20Z"'}, {'StreamArn': 'Se8255423-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table16', 'StreamLabel': '"2023-01-21T05:53:19Z"'}, {'StreamArn': 'Se351fa23-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table11', 'StreamLabel': '"2023-01-21T05:53:11Z"'}, {'StreamArn': 'Se2b67d73-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table10', 'StreamLabel': '"2023-01-21T05:53:10Z"'}, {'StreamArn': 'Se1335fe0-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table8', 'StreamLabel': '"2023-01-21T05:53:07Z"'}, {'StreamArn': 'Sdfabae73-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table5', 'StreamLabel': '"2023-01-21T05:53:05Z"'}, {'StreamArn': 'Sdec37153-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table4', 'StreamLabel': '"2023-01-21T05:53:03Z"'}, {'StreamArn': 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table2', 'StreamLabel': '"2023-01-21T05:53:01Z"'}, {'StreamArn': 'Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table0', 'StreamLabel': '"2023-01-21T05:53:00Z"'}, {'StreamArn': 'Se047a053-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table6', 'StreamLabel': '"2023-01-21T05:53:06Z"'}, {'StreamArn': 'Se5bd06b3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table14', 'StreamLabel': '"2023-01-21T05:53:15Z"'}, {'StreamArn': 'Se4d53ec3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table13', 'StreamLabel': '"2023-01-21T05:53:13Z"'}, {'StreamArn': 'Se6f119e3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table15', 'StreamLabel': '"2023-01-21T05:53:17Z"'}, {'StreamArn': 'Sde27a680-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table3', 'StreamLabel': '"2023-01-21T05:53:02Z"'}, {'StreamArn': 'Se3ed76d3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table12', 'StreamLabel': '"2023-01-21T05:53:12Z"'}, {'StreamArn': 'Se1ceb583-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table9', 'StreamLabel': '"2023-01-21T05:53:08Z"'}, {'StreamArn': 'Se95d37e3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table18', 'StreamLabel': '"2023-01-21T05:53:21Z"'}, {'StreamArn': 'Se047a053-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table6', 'StreamLabel': '"2023-01-21T05:53:06Z"'}, {'StreamArn': 'Se09746f3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table7', 'StreamLabel': '"2023-01-21T05:53:06Z"'}]
E       assert ['Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdd3c8333-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sde27a680-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdec37153-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdfabae73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se047a053-994f-11ed-9121-b5f2c5bf5f97',\n 'Se09746f3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1335fe0-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1ceb583-994f-11ed-9121-b5f2c5bf5f97',\n 'Se2b67d73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se351fa23-994f-11ed-9121-b5f2c5bf5f97',\n 'Se3ed76d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se4d53ec3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se5bd06b3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se6f119e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8255423-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8c14603-994f-11ed-9121-b5f2c5bf5f97',\n 'Se95d37e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sea454df3-994f-11ed-9121-b5f2c5bf5f97'] == ['Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sde27a680-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdec37153-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdfabae73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se047a053-994f-11ed-9121-b5f2c5bf5f97',\n 'Se047a053-994f-11ed-9121-b5f2c5bf5f97',\n 'Se09746f3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1335fe0-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1ceb583-994f-11ed-9121-b5f2c5bf5f97',\n 'Se2b67d73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se351fa23-994f-11ed-9121-b5f2c5bf5f97',\n 'Se3ed76d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se4d53ec3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se5bd06b3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se6f119e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8255423-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8c14603-994f-11ed-9121-b5f2c5bf5f97',\n 'Se95d37e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sea454df3-994f-11ed-9121-b5f2c5bf5f97']
E         At index 1 diff: 'Sdd3c8333-994f-11ed-9121-b5f2c5bf5f97' != 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97'
E         Full diff:
E           [
E            'Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97',
E         +  'Sdd3c8333-994f-11ed-9121-b5f2c5bf5f97',
E            'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97',
E            'Sde27a680-994f-11ed-9121-b5f2c5bf5f97',
E            'Sdec37153-994f-11ed-9121-b5f2c5bf5f97',
E            'Sdfabae73-994f-11ed-9121-b5f2c5bf5f97',
E         -  'Se047a053-994f-11ed-9121-b5f2c5bf5f97',
E            'Se047a053-994f-11ed-9121-b5f2c5bf5f97',
E            'Se09746f3-994f-11ed-9121-b5f2c5bf5f97',
E            'Se1335fe0-994f-11ed-9121-b5f2c5bf5f97',
E            'Se1ceb583-994f-11ed-9121-b5f2c5bf5f97',
E            'Se2b67d73-994f-11ed-9121-b5f2c5bf5f97',
E            'Se351fa23-994f-11ed-9121-b5f2c5bf5f97',
E            'Se3ed76d3-994f-11ed-9121-b5f2c5bf5f97',
E            'Se4d53ec3-994f-11ed-9121-b5f2c5bf5f97',
E            'Se5bd06b3-994f-11ed-9121-b5f2c5bf5f97',
E            'Se6f119e3-994f-11ed-9121-b5f2c5bf5f97',
E            'Se8255423-994f-11ed-9121-b5f2c5bf5f97',
E            'Se8c14603-994f-11ed-9121-b5f2c5bf5f97',
E            'Se95d37e3-994f-11ed-9121-b5f2c5bf5f97',
E            'Sea454df3-994f-11ed-9121-b5f2c5bf5f97',
E           ]

alternator_stream_tests.py:135: AssertionError
```

From
```
>       assert sorted(tables_arns) == sorted([stream['StreamArn'] for stream in total_limited_stream_responses]), \
            f"Got unexpected ARN values by list streams paged responses: {total_limited_stream_responses}"
E       AssertionError: Got unexpected ARN values by list streams paged responses: [{'StreamArn': 'Sea454df3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table19', 'StreamLabel': '"2023-01-21T05:53:23Z"'}, {'StreamArn': 'Se8c14603-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table17', 'StreamLabel': '"2023-01-21T05:53:20Z"'}, {'StreamArn': 'Se8255423-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table16', 'StreamLabel': '"2023-01-21T05:53:19Z"'}, {'StreamArn': 'Se351fa23-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table11', 'StreamLabel': '"2023-01-21T05:53:11Z"'}, {'StreamArn': 'Se2b67d73-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table10', 'StreamLabel': '"2023-01-21T05:53:10Z"'}, {'StreamArn': 'Se1335fe0-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table8', 'StreamLabel': '"2023-01-21T05:53:07Z"'}, {'StreamArn': 'Sdfabae73-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table5', 'StreamLabel': '"2023-01-21T05:53:05Z"'}, {'StreamArn': 'Sdec37153-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table4', 'StreamLabel': '"2023-01-21T05:53:03Z"'}, {'StreamArn': 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table2', 'StreamLabel': '"2023-01-21T05:53:01Z"'}, {'StreamArn': 'Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table0', 'StreamLabel': '"2023-01-21T05:53:00Z"'}, {'StreamArn': 'Se047a053-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table6', 'StreamLabel': '"2023-01-21T05:53:06Z"'}, {'StreamArn': 'Se5bd06b3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table14', 'StreamLabel': '"2023-01-21T05:53:15Z"'}, {'StreamArn': 'Se4d53ec3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table13', 'StreamLabel': '"2023-01-21T05:53:13Z"'}, {'StreamArn': 'Se6f119e3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table15', 'StreamLabel': '"2023-01-21T05:53:17Z"'}, {'StreamArn': 'Sde27a680-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table3', 'StreamLabel': '"2023-01-21T05:53:02Z"'}, {'StreamArn': 'Se3ed76d3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table12', 'StreamLabel': '"2023-01-21T05:53:12Z"'}, {'StreamArn': 'Se1ceb583-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table9', 'StreamLabel': '"2023-01-21T05:53:08Z"'}, {'StreamArn': 'Se95d37e3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table18', 'StreamLabel': '"2023-01-21T05:53:21Z"'}, {'StreamArn': 'Se047a053-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table6', 'StreamLabel': '"2023-01-21T05:53:06Z"'}, {'StreamArn': 'Se09746f3-994f-11ed-9121-b5f2c5bf5f97', 'TableName': 'user_table7', 'StreamLabel': '"2023-01-21T05:53:06Z"'}]
E       assert ['Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdd3c8333-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sde27a680-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdec37153-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdfabae73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se047a053-994f-11ed-9121-b5f2c5bf5f97',\n 'Se09746f3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1335fe0-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1ceb583-994f-11ed-9121-b5f2c5bf5f97',\n 'Se2b67d73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se351fa23-994f-11ed-9121-b5f2c5bf5f97',\n 'Se3ed76d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se4d53ec3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se5bd06b3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se6f119e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8255423-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8c14603-994f-11ed-9121-b5f2c5bf5f97',\n 'Se95d37e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sea454df3-994f-11ed-9121-b5f2c5bf5f97'] == ['Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sde27a680-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdec37153-994f-11ed-9121-b5f2c5bf5f97',\n 'Sdfabae73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se047a053-994f-11ed-9121-b5f2c5bf5f97',\n 'Se047a053-994f-11ed-9121-b5f2c5bf5f97',\n 'Se09746f3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1335fe0-994f-11ed-9121-b5f2c5bf5f97',\n 'Se1ceb583-994f-11ed-9121-b5f2c5bf5f97',\n 'Se2b67d73-994f-11ed-9121-b5f2c5bf5f97',\n 'Se351fa23-994f-11ed-9121-b5f2c5bf5f97',\n 'Se3ed76d3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se4d53ec3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se5bd06b3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se6f119e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8255423-994f-11ed-9121-b5f2c5bf5f97',\n 'Se8c14603-994f-11ed-9121-b5f2c5bf5f97',\n 'Se95d37e3-994f-11ed-9121-b5f2c5bf5f97',\n 'Sea454df3-994f-11ed-9121-b5f2c5bf5f97']
E         At index 1 diff: 'Sdd3c8333-994f-11ed-9121-b5f2c5bf5f97' != 'Sdd8bdbb3-994f-11ed-9121-b5f2c5bf5f97'
E         Full diff:
E           [
E            'Sdc9f58d3-994f-11ed-9121-b5f2c5bf5f97',
E         +  'Sdd3c8333-994f-11ed-9121-b5f2c5bf5f97',
```

I concluded that it's the **left** hand side of the equality assert which has `Sdd3c8333-994f-11ed-9121-b5f2c5bf5f97`, which means the right hand side has the duplicated ARN. And the right hand side of the equality is the concatenation of all the pages.

--- a/alternator/streams.cc
+++ b/alternator/streams.cc
@@ -140,7 +140,7 @@ namespace alternator {
 future<alternator::executor::request_return_type> alternator::executor::list_streams(client_state& client_state, service_permit permit, rjson::value request) {
     _stats.api_operations.list_streams++;
 
-    auto limit = rjson::get_opt<int>(request, "Limit").value_or(std::numeric_limits<int>::max());
+    auto limit = rjson::get_opt<int>(request, "Limit").value_or(100);
     auto streams_start = rjson::get_opt<stream_arn>(request, "ExclusiveStartStreamArn");
     auto table = find_table(_proxy, request);
     auto db = _proxy.data_dictionary();

--- a/alternator/streams.cc
+++ b/alternator/streams.cc
@@ -154,9 +154,11 @@ future<alternator::executor::request_return_type> alternator::executor::list_str
     // generate duplicates in a paged listing here. Can obviously miss things if they 
     // are added between paged calls and end up with a "smaller" UUID/ARN, but that 
     // is to be expected.
-    std::sort(cfs.begin(), cfs.end(), [](const data_dictionary::table& t1, const data_dictionary::table& t2) {
-        return t1.schema()->id().uuid() < t2.schema()->id().uuid();
-    });
+    if ((limit < cfs.size() && !table) || streams_start) {
+        std::sort(cfs.begin(), cfs.end(), [](const data_dictionary::table& t1, const data_dictionary::table& t2) {
+            return t1.schema()->id().uuid() < t2.schema()->id().uuid();
+        });
+    }
 
     auto i = cfs.begin();
     auto e = cfs.end();

--- a/alternator/streams.cc
+++ b/alternator/streams.cc
@@ -34,6 +34,7 @@
 
 #include "executor.hh"
 #include "rmw_operation.hh"
+#include "data_dictionary/data_dictionary.hh"
 
 /**
  * Base template type to implement  rapidjson::internal::TypeHelper<...>:s
@@ -144,17 +145,29 @@ future<alternator::executor::request_return_type> alternator::executor::list_str
     auto streams_start = rjson::get_opt<stream_arn>(request, "ExclusiveStartStreamArn");
     auto table = find_table(_proxy, request);
     auto db = _proxy.data_dictionary();
-    auto cfs = db.get_tables();
 
     if (limit < 1) {
         throw api_error::validation("Limit must be 1 or more");
+    }
+
+    std::vector<data_dictionary::table> cfs;
+
+    if (table) {
+        auto log_name = cdc::log_name(table->cf_name());
+        try {
+            cfs.emplace_back(db.find_table(table->ks_name(), log_name));
+        } catch (data_dictionary::no_such_column_family&) {
+            cfs.clear();
+        }
+    } else {
+        cfs = db.get_tables();
     }
 
     // # 12601 (maybe?) - sort the set of tables on ID. This should ensure we never
     // generate duplicates in a paged listing here. Can obviously miss things if they 
     // are added between paged calls and end up with a "smaller" UUID/ARN, but that 
     // is to be expected.
-    if ((limit < cfs.size() && !table) || streams_start) {
+    if (limit < cfs.size() || streams_start) {
         std::sort(cfs.begin(), cfs.end(), [](const data_dictionary::table& t1, const data_dictionary::table& t2) {
             return t1.schema()->id().uuid() < t2.schema()->id().uuid();
         });
@@ -188,14 +201,7 @@ future<alternator::executor::request_return_type> alternator::executor::list_str
         if (!is_alternator_keyspace(ks_name)) {
             continue;
         }
-        if (table && ks_name != table->ks_name()) {
-            continue;
-        }
         if (cdc::is_log_for_some_table(db.real_database(), ks_name, cf_name)) {
-            if (table && table != cdc::get_base_table(db.real_database(), *s)) {
-                continue;
-            }
-
             rjson::value new_entry = rjson::empty_object();
 
             last = i->schema()->id();

--- a/alternator/streams.cc
+++ b/alternator/streams.cc
@@ -145,19 +145,24 @@ future<alternator::executor::request_return_type> alternator::executor::list_str
     auto table = find_table(_proxy, request);
     auto db = _proxy.data_dictionary();
     auto cfs = db.get_tables();
-    auto i = cfs.begin();
-    auto e = cfs.end();
 
     if (limit < 1) {
         throw api_error::validation("Limit must be 1 or more");
     }
 
-    // TODO: the unordered_map here is not really well suited for partial
-    // querying - we're sorting on local hash order, and creating a table
-    // between queries may or may not miss info. But that should be rare,
-    // and we can probably expect this to be a single call.
+    // # 12601 (maybe?) - sort the set of tables on ID. This should ensure we never
+    // generate duplicates in a paged listing here. Can obviously miss things if they 
+    // are added between paged calls and end up with a "smaller" UUID/ARN, but that 
+    // is to be expected.
+    std::sort(cfs.begin(), cfs.end(), [](const data_dictionary::table& t1, const data_dictionary::table& t2) {
+        return t1.schema()->id().uuid() < t2.schema()->id().uuid();
+    });
+
+    auto i = cfs.begin();
+    auto e = cfs.end();
+
     if (streams_start) {
-        i = std::find_if(i, e, [&](data_dictionary::table t) {
+        i = std::find_if(i, e, [&](const data_dictionary::table& t) {
             return t.schema()->id().uuid() == streams_start
                 && cdc::get_base_table(db.real_database(), *t.schema())
                 && is_alternator_keyspace(t.schema()->ks_name())

